plutus-ledger-api/test-onchain-evaluation/Main.hs

```diff
@@ -19,46 +19,53 @@ import System.Environment (getEnv)
 import System.FilePath (isExtensionOf)
 import Test.Tasty
 import Test.Tasty.HUnit
+import Test.Tasty.Providers
+
+data SingleTest
+    = SingleTest
+        LedgerPlutusVersion
+        (Maybe (EvaluationContext, [Integer]))
+        ScriptEvaluationEvent
+
+instance IsTest SingleTest where
+    run _opts (SingleTest ver mbCtx ev) _reportProgress = case mbCtx of
+        Just (ctx, params) ->
+            evaluate
+                =<< ( pure . maybe (testPassed mempty) (testFailed . display) $
+                        checkEvaluationEvent ctx params ev
+                    )
+        Nothing -> pure . testFailed $ "missing cost parameters for " <> show ver
```

Comment:
Arguably this isn't a *test* failure, but a failure of the test construction. Maybe it should be part of the error handling in `testOneFile` :thinking:

Reply:
Yeah but cost parameters are allowed to be missing unless there are tests using it. I added a sentence to the error message "Report this as a bug against the script dumper in plutus-apps.".